### ISSUE TITLE
WebRequestPSCmdlets do not contain Response object in Exception on Mac OS X

### ISSUE THREAD

**Issue by ffeldhaus** [PowerShell/PowerShell#2113]
When running either `Invoke-WebRequest` or `Invoke-RestMethod` against a server returning an HTTP error code (e.g. 404), PowerShell 5 on Windows throws an exception which contains a response object. In PowerShell 6 on Mac OS X the response object is not available. 
## Steps to reproduce

``` powershell
Invoke-WebRequest http://httpbin.org/404 
$Error[0].Exception.Response
```
## Expected behavior

The expected behaviour is, that the exception contains a response as `HttpWebResponse` object:

``` powershell
PS C:\> $Error[0].Exception.Response

IsMutuallyAuthenticated : False
Cookies                 : {}
Headers                 : {Connection, Access-Control-Allow-Origin, Access-Control-Allow-Credentials, Content-Length...}
SupportsHeaders         : True
ContentLength           : 233
ContentEncoding         : 
ContentType             : text/html
CharacterSet            : ISO-8859-1
Server                  : nginx
LastModified            : 8/29/2016 12:47:17 AM
StatusCode              : NotFound
StatusDescription       : NOT FOUND
ProtocolVersion         : 1.1
ResponseUri             : http://httpbin.org/404
Method                  : GET
IsFromCache             : False
```
## Actual behavior

``` powershell
PS /> $Error[0].Exception.Response
PS /> 
```
## Environment data

<!-- provide the output of $PSVersionTable -->

``` powershell
Name                           Value                                                                                                                                                                                                         
----                           -----                                                                                                                                                                                                         
PSVersion                      6.0.0-alpha                                                                                                                                                                                                   
PSEdition                      Core                                                                                                                                                                                                          
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                       
BuildVersion                   3.0.0.0                                                                                                                                                                                                       
GitCommitId                    v6.0.0-alpha.9                                                                                                                                                                                                
CLRVersion                                                                                                                                                                                                                                   
WSManStackVersion              3.0                                                                                                                                                                                                           
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                           
SerializationVersion           1.1.0.1 
```

**Comment by Jaykul** [comment]
This affects PowerShell Core everywhere, not just OS X.

The problem is that `Invoke-WebRequest` on PowerShell Core uses `HttpClient` and thus throws a `System.Net.Http.HttpRequestException` instead of a `System.Net.WebException` ...

**Comment by vors** [comment]
cc @Francisco-Gamino

**Comment by daxian-dbw** [comment]
Fixed via #3201
